src/server/tests/test_stac_api_collection_search.py

```diff
@@ -70,6 +71,18 @@ def test_hint(mock_apis):
         "\nitem_collection = search.item_collection()"
     )
     for i, result in enumerate(results):
+        assert isinstance(result, CollectionMetadata)
         assert result.hint
         if i == 0:
             assert result.hint.strip() == expected_hint.strip()
+
+
+def test_api_error():
+    # a search against an API that doesn't exist
+    base_search = STACAPICollectionSearch(
+        base_url="http://nope",
+    )
+
+    results = list(base_search.get_collection_metadata())
+    for result in results:
+        assert isinstance(result, FederatedSearchError)
```

Comment:
Feel free to ignore, but minimally, you can avoid converting to a list and simply do this:

```suggestion
    for result in base_search.get_collection_metadata():
        assert isinstance(result, FederatedSearchError)
```

Perhaps more appropriately though, you want to check that _all_ instances are errors (although, in this case it would appear that you should get only 1 instance):

```suggestion
    assert all(
        isinstance(result, FederatedSearchError)
        for result in base_search.get_collection_metadata()
    )
```

If you were expecting multiple values, then performing an assert in a loop would identify only the _first_ result that's not an error, thus preventing you from seeing _all_ results that are not errors. However, for pytest to display a more helpful output in such a case, I believe you would actually want to use a list within `all`, otherwise I think pytest would show you the generator instance in the error output, not individual results, thus making it impossible to see the non-error items. Anyway, I'm heading down a rabbit hole, so I'll stop. Just some food for thought.

Reply:
In this case the result should always have a length of one, but in case that changes in the future it would be smart to check all elements. Thanks for the suggestion!